lib/spack/spack/binary_distribution.py

```diff
@@ -926,41 +834,60 @@ def _fetch_spec_from_mirror(spec_url):
         tp.terminate()
         tp.join()
 
-    for fetched_spec in fetched_specs:
-        db.add(fetched_spec, None)
-        db.mark(fetched_spec, "in_buildcache", True)
+    # Recursively lower the Specs into spec-dict (+ record) form
```

Comment:
Why are you reinventing the database format here? Better to leave it as it was so we don't have to do double maintenance.

Reply:
The intention was to separate the `index.json` code paths here from the `Database` implementation, without changing the format itself (for compatibility with older Spack). This would allow the `index.json` format to evolve separately from the far more complete `Database`.

Delaying this change for a later PR.